### PR TITLE
tests: Bluetooth: Audio: Remove uses of K_FOREVER in syswg for TX

### LINKED ISSUE #80682: Bluetooth: Audio: BSIM tests use `K_FOREVER` when allocating TX buffers in system workqueue
**Describe the bug**
Several of the audio BSIM tests are allocating buffers as `net_buf_alloc(&tx_pool, K_FOREVER);` in the system workqueue, which isn't allowed. 

**To Reproduce**
Build the audio BSIM test and run any of the test that send data, e.g. the `broadcast_source`

**Expected behavior**
The tests should not use `K_FOREVER` in the system workqueue as that can result in lost TXs if the buffers are not immediately available. 

**Impact**
The tests are subject to timing changes where timing may change how they send. For most tests it is "just" a lot of warning logs like `<wrn> net_buf: net_buf_alloc_len: Timeout discarded. No blocking in syswq`

**Logs and console output**
```
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_iso: hci_le_big_complete: BIG[0] 0x8258e00 completed, status 0x00 
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_iso: bt_iso_connected: 0x8258a80
d_00: @00:00:00.017327  [00:00:00.017,303] <wrn> net_buf: net_buf_alloc_len: Timeout discarded. No blocking in syswq
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_iso: bt_iso_chan_set_state_debug: chan 0x825d7c0 iso 0x8258a80 connecting -> connected
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_bap_broadcast_source: broadcast_source_iso_connected: stream 0x8251080 ep 0x825f8e0
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_bap_broadcast_source: broadcast_source_set_ep_state: ep 0x825f8e0 id 0x00 enabling -> streaming
d_00: @00:00:00.017327  Stream 0x8251080 started
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_iso: bt_iso_connected: 0x8258b5c
d_00: @00:00:00.017327  [00:00:00.017,303] <wrn> net_buf: net_buf_alloc_len: Timeout discarded. No blocking in syswq
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_iso: bt_iso_chan_set_state_debug: chan 0x825d850 iso 0x8258b5c connecting -> connected
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_bap_broadcast_source: broadcast_source_iso_connected: stream 0x82510c4 ep 0x825fa4c
d_00: @00:00:00.017327  [00:00:00.017,303] <dbg> bt_bap_broadcast_source: broadcast_source_set_ep_state: ep 0x825fa4c id 0x00 enabling -> streaming
```

**Environment (please complete the following information):**
 - Commit SHA or Version used: 124aae3876842154964726ba6e89565040648f9a

**Additional context**

### PR DESCRIPTION
**Pull request by Thalley**
Several tests were using K_FOREVER when allocating the buffer for TX in the system workqueue, which is illegal behavior.

The solution chosen was to create a TX thread to handle TX, similar to the solution used in the audio shell and some sample applications.

This way we can continue to use K_FOREVER when allocting buffers and it will always be done in a round-robin fashion while TXing as much as possible, by always enqueuing all the buffers with mock data.

Since this works for all streams (both broadcast and unicast), it was obvious to use the same implementation for all tests, and thus cleaning up the tests a bit and more them more similar.

fixes https://github.com/zephyrproject-rtos/zephyr/issues/80682

### CODE REVIEW

File: tests/bsim/bluetooth/audio/src/bap_stream_tx.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+int stream_tx_unregister(struct bt_bap_stream *bap_stream);
+
+/**
+ * @brief Test if the provided stream has been configured for TX
+ *
+ * @param bap_stream The stream to test for TX support
+ *
+ * @retval true if it has been configured for TX, and false if not
+ */
+bool stream_tx_can_send(const struct bt_bap_stream *stream);
```
Comment (rruuaanng) [inline tests/bsim/bluetooth/audio/src/bap_stream_tx.h:61]: Why is it `can_tx`, not `is_tx` or `is_conf`.

Comment (Thalley) [reply tests/bsim/bluetooth/audio/src/bap_stream_tx.h:61]: To follow the `stream_tx` prefit it sounds weird to have `stream_tx_is_send` or `stream_tx_is_conf_send` as `stream_tx` refers to the module, and not the stream

File: tests/bsim/bluetooth/audio/src/bap_stream_rx.c [hunk 2]
```diff
@@ -35,29 +35,29 @@ void bap_stream_rx_recv_cb(struct bt_bap_stream *stream, const struct bt_iso_rec
 
 	if (test_stream->rx_cnt > 0U && info->ts == test_stream->last_info.ts) {
 		log_stream_rx(stream, info, buf);
-		FAIL("Duplicated timestamp received: %u\n", test_stream->last_info.ts);
+		FAIL("Duplicated timestamp received: %u", test_stream->last_info.ts);
```
Comment (kruithofa) [inline tests/bsim/bluetooth/audio/src/bap_stream_rx.c:38]: The `FAIL` macro does not add a CR-LF, so I don't think you want to do this change
We should probably fix this at other places as well

File: tests/bsim/bluetooth/audio/src/bap_stream_rx.c [hunk 3]
```diff
@@ -13,14 +13,14 @@
 #include "common.h"
 #include <string.h>
 
-LOG_MODULE_REGISTER(stream_tx, LOG_LEVEL_INF);
+LOG_MODULE_REGISTER(stream_rx, LOG_LEVEL_INF);
```
Comment (kruithofa) [inline tests/bsim/bluetooth/audio/src/bap_stream_rx.c:16]:
```suggestion
LOG_MODULE_REGISTER(bap_stream_rx, LOG_LEVEL_INF);
```

Not a required change but the other modules use the complete module-name when registering with the logging system.

File: tests/bsim/bluetooth/audio/src/bap_unicast_client_test.c [hunk 4]
```diff
@@ -106,6 +102,16 @@ static void stream_started(struct bt_bap_stream *stream)
 {
 	printk("Started stream %p\n", stream);
 
+	if (stream_tx_can_send(stream)) {
```
Comment (kruithofa) [inline tests/bsim/bluetooth/audio/src/bap_unicast_client_test.c:105]: question: Do I understand it correctly that when `stream_tx_can_send` is false we do not actually start streaming for this `stream`?
If my understanding is correct then probably we shouldn't set the `flag_stream_started`

Comment (Thalley) [reply tests/bsim/bluetooth/audio/src/bap_unicast_client_test.c:105]: Streams are unidirectional. For RX-only streams `stream_tx_can_send` will return false, but the stream is still started in the sense that the ASE is in the streaming state

**Review by kruithofa — DISMISSED** [review]

**Review by cvinayak — DISMISSED** [review]

**Comment by kartben** [comment]
@Thalley needs a rebase

**Review by kruithofa — APPROVED** [review]

**Review by cvinayak — APPROVED** [review]

**Pull request merged**